Add links to app

iamvery opened the pull request.

iamvery committed:
Install dependencies for persistence using rom.rb

iamvery committed:
Configure rom in app

iamvery committed:
Rough in links feature test

iamvery committed:
Add view for links index

Pakyow is _view first_! :dizzy:

iamvery committed:
Add route to bind links to view

This creates a link list item for each value found in the database and
binds each's URL to the anchor tag. But it should be noted that the href
attribute is _not_ bound. We have to do that explicitly! Coming up next.

iamvery committed:
Ensure href is updated for bound links

bryanp commented on app/lib/routes.rb:
```diff
@@ -3,5 +3,13 @@
+      links = Pakyow::Config.app.rom.relation(:links).to_a
+
+      view.scope(:link).apply(links) do |view, datum|
+        view.prop(:url).attrs[:href] = datum.fetch(:url)
```
I'd do this in a binding instead, then the view logic could simply be:

``` ruby
view.scope(:link).apply(links)
```

iamvery replied:
I tried it out! It does look a lot better, though it surprised me that when a `binding :foo` exists in the bindings, you loose the default behavior, i.e. content binding.

Does that make sense?

iamvery committed:
Replace awkward manual attribute assignment with binding

h/t @bryanp

This reads a little better. Turns out "bindings" are sort of like
callbacks that are invoked whenever binding happens to the view. :dizzy:
